src/main/java/com/mraof/minestuck/item/weapon/OnHitEffect.java

```diff
@@ -314,6 +340,41 @@ static OnHitEffect enemyPotionEffect(Supplier<MobEffectInstance> effect)
 		return (stack, target, attacker) -> target.addEffect(effect.get());
 	}
 	
+	//Nearly identical to Sweep, but it applies an optional array of effects
+	static OnHitEffect sweepMultiEffect(OnHitEffect... effects)
+	{
+		return (stack, target, attacker) -> {
+			if(attacker instanceof Player playerAttacker)
+			{
+				boolean slowMoving = (double) (playerAttacker.walkDist - playerAttacker.walkDistO) < (double) playerAttacker.getSpeed();
+				boolean lastHitWasCrit = ServerEventHandler.wasLastHitCrit(playerAttacker);
+				if(slowMoving && !lastHitWasCrit && playerAttacker.isOnGround()
+						&& playerAttacker.getAttackStrengthScale(0) >= 1)
+				{
+					float attackDamage = (float) playerAttacker.getAttribute(Attributes.ATTACK_DAMAGE).getValue();
+					float sweepEnchantMod = 1.0F + EnchantmentHelper.getSweepingDamageRatio(playerAttacker) * attackDamage;
+					
+					for(LivingEntity livingEntity : playerAttacker.level.getEntitiesOfClass(
+							LivingEntity.class, target.getBoundingBox().inflate(2.0D, 0.25D, 2.0D)))
+					{
+						if(livingEntity != playerAttacker && !playerAttacker.isAlliedTo(livingEntity)
+								&& (!(livingEntity instanceof ArmorStand) || !((ArmorStand) livingEntity).isMarker())
+								&& playerAttacker.distanceToSqr(livingEntity) < 9.0D)
+						{
+							if(livingEntity != target)
+								livingEntity.hurt(DamageSource.playerAttack(playerAttacker), sweepEnchantMod);
+							for(OnHitEffect effect : effects)
+								effect.onHit(stack, livingEntity, attacker);
+						}
+					}
+					
+					playerAttacker.level.playSound(null, playerAttacker.getX(), playerAttacker.getY(), playerAttacker.getZ(), SoundEvents.PLAYER_ATTACK_SWEEP, playerAttacker.getSoundSource(), 1.0F, 1.0F);
+					playerAttacker.sweepAttack();
+				}
+			}
+		};
+	}
```

Comment:
The function could do with some transformations to improve readability. To start, I recommend [inverting](https://youtu.be/CFRhGnuXG-4) the first two if-statements and making an early return. After that, it could be useful to extract some part of the code inside the for-loop or if-statement conditions.

Reply:
Inversions were made, I did not feel extractions were necessary. Inversions to reduce nesting were made to SuspicionEffect.applyEffectTick(LivingEntity, int) as well.